src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -251,6 +263,120 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
+                    // 1. All the fields from the target node to perform insertion on the target entity,
+                    // 2. Fields from the linking node which are not a foreign key reference to source or target node. This is needed to perform
+                    // an insertion in the linking table. For the foreign key columns in linking table, the values are derived from the insertions in the
+                    // source and the target table. For the rest of the columns, the value will be provided via a field exposed in the sourceTargetLinkingNode.
+                    foreach (FieldDefinitionNode fieldInLinkingNode in fieldsInLinkingNode)
+                    {
+                        string fieldName = fieldInLinkingNode.Name.Value;
+                        if (!referencingColumnNames.Contains(fieldName))
+                        {
+                            if (fieldNamesInTarget.Contains(fieldName))
+                            {
+                                // The fieldName can represent a column in the targetEntity or a relationship.
+                                // The fieldName in the linking node cannot conflict with any of the
+                                // existing field names (either column name or relationship name) in the target node.
+                                bool doesFieldRepresentAColumn = sqlMetadataProvider.TryGetBackingColumn(targetEntityName, fieldName, out string? _);
+                                string infoMsg = $"Cannot use field name '{fieldName}' as it conflicts with one of the other field's name in the entity: {targetEntityName}. ";
+                                string actionableMsg = doesFieldRepresentAColumn ?
+                                    $"Consider using the 'mappings' section of the {targetEntityName} entity configuration to provide some other name for the field: '{fieldName}'." :
+                                    $"Consider using the 'relationships' section of the {targetEntityName} entity configuration to provide some other name for the relationship: '{fieldName}'.";
+                                throw new DataApiBuilderException(
+                                    message: infoMsg + actionableMsg,
+                                    statusCode: HttpStatusCode.ServiceUnavailable,
+                                    subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
+                            }
+                            else
+                            {
+                                fieldsInSourceTargetLinkingNode.Add(fieldInLinkingNode);
+                            }
+                        }
+                    }
+
+                    // Store object type of the linking node for (sourceEntityName, targetEntityName).
+                    NameNode sourceTargetLinkingNodeName = new(GenerateLinkingNodeName(objectTypes[sourceEntityName].Name.Value, objectTypes[targetEntityName].Name.Value));
```

Comment:
nit: line too long, constructor params can be split into multiple lines to keep line limit = 100